src/main/java/sk/hfa/blog/web/BlogController.java

```diff
@@ -9,52 +10,50 @@
 import sk.hfa.blog.domain.BlogArticle;
 import sk.hfa.blog.domain.BlogArticleDto;
 import sk.hfa.blog.services.interfaces.IBlogService;
-import sk.hfa.blog.web.domain.requestbodies.BlogArticleRequest;
 import sk.hfa.blog.web.domain.responsebodies.BlogArticleMessageResource;
 import sk.hfa.blog.web.domain.responsebodies.BlogArticlePageMessageResource;
-import sk.hfa.images.services.interfaces.IImageService;
 import sk.hfa.web.domain.responsebodies.DeleteEntityMessageResource;
 import sk.hfa.web.domain.responsebodies.MessageResource;
 
-import java.util.Objects;
+import javax.validation.Valid;
 
 @Slf4j
 @RestController
 @RequestMapping(path = "/api/blog")
 public class BlogController {
 
-    private final IImageService imageService;
     private final IBlogService blogService;
 
-    public BlogController(IImageService imageService, IBlogService blogService) {
-        this.imageService = imageService;
+    public BlogController(IBlogService blogService) {
         this.blogService = blogService;
     }
 
     @PreAuthorize("hasRole('ADMIN')")
-    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<MessageResource> createBlogArticle(@RequestBody BlogArticleRequest request) {
-        String operation = "was successfully created.";
-        String message = "Creating a new blog article.";
-
-        if (!Objects.isNull(request.getBlogArticle().getId())) {
-            imageService.deleteBlogArticleImage(request.getBlogArticle().getId());
-            message = "Updating an existing blog article.";
-            operation = "was successfully updated.";
-        }
+    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<MessageResource> createBlogArticle(@Valid BlogArticleDto request) throws JsonProcessingException {
+        log.info("Creating a new blog article.");
+        BlogArticle blogArticle = blogService.save(request);
+        MessageResource responseBody = new BlogArticleMessageResource(blogArticle);
+        log.info("The blog article with the ID: [" + blogArticle.getId() + "] was successfully created.");
+        return ResponseEntity.ok(responseBody);
+    }
 
-        log.info(message);
-        BlogArticle blogArticle = blogService.save(BlogArticle.build(request.getBlogArticle()));
-        MessageResource responseBody = new BlogArticleMessageResource(BlogArticleDto.build(blogArticle, false));
-        log.info("The blog article with the ID: [" + blogArticle.getId() + "] " + operation);
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping(produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<MessageResource> updateBlogArticle(@Valid  BlogArticleDto request) throws JsonProcessingException {
```

Comment:
pred PR to sformatujme nech niesu potom problemy v dalsich PR

Reply:
Viem ze sme to tak nedodrziavali ani predtym ale mali by sme zacat objekty davat ako final